### ISSUE TITLE
Question - What is the purpose of this piece of Code in densenet.py?

### ISSUE THREAD

**Issue by gokulanv** [bamos/densenet.pytorch#21]
Learning CNN architectures. Can you please tell what this piece of code does and why we are doing this? I could not relate it with the paper.

for m in self.modules():
            if isinstance(m, nn.Conv2d):
                n = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
                m.weight.data.normal_(0, math.sqrt(2. / n))
            elif isinstance(m, nn.BatchNorm2d):
                m.weight.data.fill_(1)
                m.bias.data.zero_()
            elif isinstance(m, nn.Linear):
                m.bias.data.zero_()

**Comment by qiangwang57** [comment]
Those lines are for initialising weight and bias.
